doc/efa-env-var.md

```diff
@@ -0,0 +1,105 @@
+# EFA Cheatsheet
+
+## 1. Settings via environment variables
+
+For optimized performance, you may need to set additional environment variables depending on the
+versions of your libfabric.
+
+<table>
+   <thead>
+      <th>Setting</th>
+      <th>Description</th>
+   </thead>
+   <tr>
+      <td><code>FI_EFA_USE_HUGE_PAGE=0</code></td>
+      <td>Set to 0 when you see <code>os.fork()</code> causes <code>OSError: Cannot allocate
+      memory</code>. Typically happen by multi-process PyTorch data loader. Disabling huge page
+      causes minor performance hit, but it's needed to prevent fork fails due to the operating
+      system running out of huge pages.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_FORK_SAFE=1</code></td>
+      <td>Not needed for kernel>=5.15. Still fine to set it though no effect. See
+      <a href=https://github.com/ofiwg/libfabric/pull/9112>ref</a>.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_USE_DEVICE_RDMA=1</code></td>
+      <td>Do not set for libfabric>=1.18.0 and aws-ofi-nccl>=1.7.0. It's not harmful to set it on
+      p4/p5 instances with the newer software, but you just don't have to set it.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_ENABLE_SHM_TRANSFER=1</code></td>
+      <td>Not needed. This is really a no-op, as aws-ofi-nccl defaults to enable SHMEM.</td>
+   </tr>
+   <tr>
+      <td><code>FI_PROVIDER=efa</code></td>
+      <td>Use for aws-ofi-nccl<=1.5.0 AND p4/p5 instances.</td>
```

Comment:
Wouldn't this be needed for all instances types with EFA?: p3 and g5 as well.

Reply:
Will be corrected to include all EFA-enabled instance types.